source/uwp/Visualizer/ViewModel/GenericDocumentViewModel.cs

```diff
@@ -117,15 +120,58 @@ protected async Task LoadFromFileAsync(IStorageFile file, string token, bool ass
             }
         }
 
+        private async Task CreateMessageDialog(string text)
+        {
+            var dontWait = new MessageDialog(text);
+            dontWait.CancelCommandIndex = 0;
+            await dontWait.ShowAsync();
+        }
+
+        private async Task SaveNewFile()
+        {
+            var savePicker = new FileSavePicker();
+            savePicker.SuggestedStartLocation = PickerLocationId.DocumentsLibrary;
+            savePicker.FileTypeChoices.Add("Plain Text", new List<string>() { ".json" });
+            savePicker.SuggestedFileName = "NewAdaptiveCard.json";
+
+            File = await savePicker.PickSaveFileAsync();
+            if (File != null)
+            {
+                CachedFileManager.DeferUpdates(File);
+                await FileIO.WriteTextAsync(File, File.Name);
+                FileUpdateStatus status = await CachedFileManager.CompleteUpdatesAsync(File);
+                if (status == FileUpdateStatus.Complete)
+                {
+                    this.Name = File.Name;
+                    await CreateMessageDialog("File " + File.Name + " was saved");
+                }
+                else
+                {
+                    await CreateMessageDialog("File " + File.Name + " couldn't be saved");
+                }
+            }
+            else
+            {
+                await CreateMessageDialog("Save new file was cancelled");
+            }
+        }
+
         public async Task SaveAsync()
         {
-            try
+            if (File == null)
             {
-                await FileIO.WriteTextAsync(File, Payload);
+                await SaveNewFile();
             }
-            catch (Exception ex)
+            else
             {
-                var dontWait = new MessageDialog(ex.ToString()).ShowAsync();
+                try
+                {
+                    await FileIO.WriteTextAsync(File, Payload);
+                }
+                catch (Exception ex)
```

Comment:
We should figure out why we are getting exceptions in the "Save" workflow to begin with here (one of them is issue #2844)

Reply:
I couldn't repro the System.UnauthorizedAccessException. Have you tried building the app in your local machine and check if the same exception pop ups. Otherwise we should check if it's an issue with the app in the store

---
In reply to: [389920350](https://github.com/microsoft/AdaptiveCards/pull/3825#discussion_r389920350) [](ancestors = 389920350)

Reply:
I could repro the issue now and I think it was a big oversight by us
https://social.msdn.microsoft.com/Forums/en-US/a32d16fd-7855-4fce-9091-8067f712a1b6/access-denied-on-windowsstoragefileiowritetextasync-to-write-a-text-file-in-a-local-application?forum=winappswithhtml5

It looks like we can't save on the app folder, during the build process we copy some files to the app folder (i.e. AppX\Samples\Agenda.json), this means that all cards inside the app folder can we overwritten as we don't have permission by UWP's design. I'm not sure what scenarios we use the Save function for as any cards loaded from another location or newly created won't have a token and they are not loaded on app startup